### DISCUSSION TITLE
İdam mahkumları ve şapkaları sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#83] [Zeka Sorusu] [answered]
![s008](https://github.com/sonsuzus/sonsuzus/assets/52765796/bd6a2ae3-8646-4e30-a444-a867f63fe8b9)

Yukarıdaki resimde 4 kişi var ve hepsi boynundan toprağa gömülmüş, dönemiyorlar. Dolayısıyla hepsi sadece önüne bakıyor.
A ile B arasında da kalın bir duvar var, arkası gözükmüyor.
Hepsi biliyor ki, toplamda 4 kişiler ve 2 siyah 2 beyaz şapka giyiyorlar. Ama hiç biri kendisinin ne renk şapka giydiğini bilmiyor. Ve her biri diğer 3 kişinin nerede gömülü olduğunu biliyor.
Öldürülmemeleri için içlerinden herhangi birinin cellada kendi şapkasının rengini doğru olarak söylemesi lazım. Eğer yanlış söylerse cellat hepsini öldürecek.
Elbette birbirleriyle konuşmaları yasak ve sadece 10 dakikaları var..
Ama aradan 1 dakika geçtikten sonra birisi çıkıp %100 emin olduğunu söyleyerek kendi şapkasının rengini tahmin ediyor ve biliyor.
SORU: Hangisi bildi ve nasıl %100 emin olabiliyor?

**Comment by HabilCcabbar** [comment, marked as answer]
_**C**_

C şöyle düşünüyor. En çok kişiyi D görüyor. Eğer önünde gördüğü kişilerin ikisinin de şapkası aynı renk olsa, kendi şapkasının rengini bulabilirdi. Yani B ve C ikisi de beyaz olsa, D'nin siyah olacağı kesindi. D konuşmadığına göre demek ki B ve C'nin şapkaları farklı renk. C, B'nin şapkasının beyaz renk olduğunu gördüğü için kendi şapkasının siyah renk olduğundan emin oluyor ve söylüyor.
